pkg/pgmodel/labels.go

```diff
@@ -112,14 +113,16 @@ func getStr(labels []prompb.Label) (string, error) {
 	return builder.String(), nil
 }
 
+// labelProtosToLabels converts promb label-pairs to Label types. Labels type are created and stored in-memory
+// if they do not exist. If they exist, they are simply returned thereby preventing memory re-allocations.
 func labelProtosToLabels(labelPairs []prompb.Label) (*Labels, string, error) {
-	str, err := getStr(labelPairs)
+	str, err := getLabelsStringRepresentation(labelPairs)
 	if err != nil {
 		return nil, "", err
 	}
 	labels := GetLabels(str)
 	if labels == nil {
-		labels = new(Labels)
+		labels = &Labels{}
```

Comment:
For maintaining uniformity across the code. Some places have `new()` and other have address notations. let's go with address notations since they are widely used in our repo and other places as well.